3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/EnrichmentManager.scala

```diff
@@ -508,6 +517,8 @@ object EnrichmentManager {
     // Assemble array of contexts prepared by built-in enrichments
     val preparedDerivedContexts = List(uaParser).collect {
       case Success(Some(context)) => context
+    } ++ List(openCageContext).collect {
+      case Success(Some(context)) => context
```

Comment:
At the moment, if any of the enrichments fails, the event should go to bad rows. It's not the case here. You can check the 3rd point [here](https://github.com/snowplow/snowplow/tree/4009_yauaa_enrichment/3-enrich#2c-enrichmentmanager) for how to do it.